### PR TITLE
[JIRA] (ACDM-969) Improvement: Possibility to plug new enrolment managers

### PR DESCRIPTION
**Pull request by nadiramin**
Closes ACDM-969.

### CODE REVIEW

File: src/main/java/org/fenixedu/academic/domain/studentCurriculum/StudentCurricularPlanEnrolment.java [hunk 1]
```diff
@@ -380,31 +381,45 @@ protected boolean isResponsiblePersonCoordinator() {
     // -------------------
 
     static public StudentCurricularPlanEnrolment createManager(final EnrolmentContext enrolmentContext) {
+        return ENROLMENT_MANAGER_FACTORY.get().createManager(enrolmentContext);
+    }
 
-        if (enrolmentContext.isNormal()) {
+    static public Supplier<EnrolmentManagerFactory> ENROLMENT_MANAGER_FACTORY = () -> new EnrolmentManagerFactory() {
```
Comment (jcarvalho) [inline src/main/java/org/fenixedu/academic/domain/studentCurriculum/StudentCurricularPlanEnrolment.java]: Don't publicly expose an internal field, keep it `private` and expose a setter (which checks that the new factory is not `null`).

**Comment by legidio** [comment]
Will do.

As a side note, maybe the same suggested checks should be added in this extension point:

https://github.com/FenixEdu/fenixedu-academic/blob/master/src/main/java/org/fenixedu/academic/domain/student/curriculum/Curriculum.java#L162

**Comment by pedrosan7os** [comment]
Accepted suggestion: a49e8fe4d52fe11bf51062af5b3b378159ee8afe

**Comment by legidio** [comment]
Done.

Applied same rationale, with the adition of also testing for the Supplier's result, obtained from the get() method.

**Pull request merged**